app/models/task.py

```diff
@@ -1,5 +1,37 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from sqlalchemy import ForeignKey
+from datetime import datetime
+from typing import TYPE_CHECKING, Optional
+if TYPE_CHECKING:
+    from .goal import Goal
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]] = mapped_column(nullable=True)
```

Comment:
Using `Optional[]` is all that we need to indicate that a field is nullable so we don't need to also include `mapped_colum`n with `nullable=True` (ilke how you did it for `goal_id` below on line 14).

```suggestion
    completed_at: Mapped[Optional[datetime]] 
```